src/Base/Fields/HasMany.php

```diff
@@ -207,19 +196,24 @@ private function formattedEntry($item)
                 ->formattedResponse();
         }
 
-        if ($this->isNestable() && $item->children->count()) {
-            foreach ($item->children->sortBy($this->getSortableColumn()) as $child) {
-                $entry['children'][] = $this->formattedEntry($child);
+        if ($this->isNestable()) {
+            $tree = [
+                'id'       => (string)$item->id,
+                'text'     => $item->name,
+                'parent'   => $item->parent_id ? (string)$item->parent_id : '#',
+                'children' => [],
+            ];
+
+            if (get_class($item) === config('laradium.menu_item_class', '\Laradium\Laradium\Models\MenuItem')) {
```

Comment:
```suggestion
            if (get_class($item) === config('laradium.menu_item_class', \Laradium\Laradium\Models\MenuItem::class)) {
```

Just my personal preference, you can ignore this